extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetInstructions.java

```diff
@@ -822,6 +882,35 @@ public Builder useDictionary(final String columnName, final boolean useDictionar
             return this;
         }
 
+        /**
+         * For reading, provides a mapping between a Deephaven column name and a parquet column by field id. This allows
+         * resolving a parquet column where the physical "parquet column name" may not be known apriori by the caller.
+         * This may happen in cases where the parquet file is managed by a higher-level schema that has the concept of a
+         * "field id"; for example, Iceberg. As <a href=
+         * "https://github.com/apache/parquet-format/blob/apache-parquet-format-2.10.0/src/main/thrift/parquet.thrift#L456-L459">documented
+         * in the parquet format</a>:
+         *
+         * <pre>
+         *     When the original schema supports field ids, this will save the original field id in the parquet schema
+         * </pre>
+         *
+         * In the case where a field id mapping is provided but no matching parquet column is found, the column will not
+         * be inferred; and in the case where it's explicitly included as part of a
+         * {@link #setTableDefinition(TableDefinition)}, the resulting column will contain the appropriate default
+         * ({@code null}) values.
+         *
+         * <p>
+         * This field is not typically configured by end users.
+         *
+         * <p>
+         * For writing, this is not currently supported.
+         */
+        public Builder setFieldId(final String columnName, final int fieldId) {
+            final ColumnInstructions ci = getColumnInstructions(columnName);
```

Comment:
We can add a `Set` or `HashMap` here to check if the field ID is already being used fro some other column and then maybe log the name of the existing column as well which is using the id.

Reply:
Actually, it's completely legit (I think) to re-use fieldId from this context - the unique key is only "columnName". I added an explicit test case for it.

Reply:
I couldn't find any resources to confirm, but this does feel incorrect to me, having two columns with same field ID. For example, if we get a field ID by Iceberg, it would expect a single column, right?